library/std/src/sys/unix/net.rs

```diff
@@ -430,6 +430,17 @@ impl Socket {
         Ok(passcred != 0)
     }
 
+    #[cfg(all(target_os = "freebsd", freebsd13))]
+    pub fn set_passcred(&self, passcred: bool) -> io::Result<()> {
+        setsockopt(self, libc::AF_LOCAL, libc::LOCAL_CREDS_PERSISTENT, passcred as libc::c_int)
+    }
+
+    #[cfg(all(target_os = "freebsd", freebsd13))]
+    pub fn passcred(&self) -> io::Result<bool> {
+        let passcred: libc::c_int = getsockopt(self, libc::AF_LOCAL, libc::LOCAL_CREDS_PERSISTENT)?;
```

Comment:
Why use LOCAL_CREDS_PERSISTENT for FreeBSD, but LOCAL_CREDS for NetBSD?

Reply:
Because LOCAL_CREDS_PERSISTENT is used in conjunction with `sockcred2` as opposed to LOCAL_CREDS and `sockcred` which does not get the process id on FreeBSD. NetBSD does not have the same history regarding this feature.